Build failure: cnijfilter2

## Steps To Reproduce

Steps to reproduce the behavior:

Try to switch to current nixpkgs-unstable with printing for canon enabled:

```nix
services.printing = {
	enable = true;
	drivers = [ pkgs.cnijfilter2 ];
};
```



## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```sh
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gxlhvx64b36rz0w1ff8g0d87pvdvf7mg-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/dkyw3qzna54021hfv7p43f2ckaxhvd55-get_protocol.patch
patching file lgmon3/src/cnijlgmon3.c
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
substituteStream() in derivation cnijfilter2-6.40: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'src/Makefile.am')
processing .
Creating ./aclocal.m4 ...
Running glib-gettextize...  Ignore non-fatal messages.
Copying file po/Makefile.in.in

Please add the files
  codeset.m4 gettext.m4 glibc21.m4 iconv.m4 isc-posix.m4 lcmessage.m4
  progtest.m4
from the /nix/store/rb870mk51kxlxq9crkldzs8n4vwr9sk4-glib-2.82.1/share/aclocal directory to your autoconf macro directory
or directly to your aclocal.m4 file.
You will also need config.guess and config.sub, which you can get from
ftp://ftp.gnu.org/pub/gnu/config/.

Making ./aclocal.m4 writable ...
Running libtoolize...
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.in,
libtoolize: and rerunning libtoolize and aclocal.
Running aclocal  ...
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
Running autoheader...
autoheader: warning: autoconf input should be named 'configure.ac', not 'configure.in'
autoheader: warning: WARNING: Using auxiliary files such as 'acconfig.h', 'config.h.bot'
autoheader: WARNING: and 'config.h.top', to define templates for 'config.h.in'
autoheader: WARNING: is deprecated and discouraged.
autoheader: 
autoheader: WARNING: Using the third argument of 'AC_DEFINE_UNQUOTED' and
autoheader: WARNING: 'AC_DEFINE' allows one to define a template without
autoheader: WARNING: 'acconfig.h':
autoheader: 
autoheader: WARNING:   AC_DEFINE([NEED_FUNC_MAIN], 1,
autoheader: 		[Define if a function 'main' is needed.])
autoheader: 
autoheader: WARNING: More sophisticated templates can also be produced, see the
autoheader: WARNING: documentation.
Running automake --gnu  ...
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:2: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.in:2: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.in:24: installing './compile'
configure.in:27: installing './config.guess'
configure.in:27: installing './config.sub'
configure.in:2: installing './install-sh'
configure.in:2: installing './missing'
src/Makefile.am: installing './depcomp'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
src/Makefile.am:9: warning: 'CFLAGS' is a user variable, you should not override it;
src/Makefile.am:9: use 'AM_CFLAGS' instead
Running autoconf ...
configure.in:2: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
aclocal.m4:9977: AM_INIT_AUTOMAKE is expanded from...
configure.in:2: the top level
configure.in:4: warning: 'AM_CONFIG_HEADER': this macro is obsolete.
configure.in:4: You should use the 'AC_CONFIG_HEADERS' macro instead.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
aclocal.m4:10293: AM_CONFIG_HEADER is expanded from...
configure.in:4: the top level
configure.in:27: warning: The macro 'AM_PROG_LIBTOOL' is obsolete.
configure.in:27: You should run autoupdate.
aclocal.m4:123: AM_PROG_LIBTOOL is expanded from...
configure.in:27: the top level
configure.in:34: warning: The macro 'AC_HEADER_STDC' is obsolete.
configure.in:34: You should run autoupdate.
./lib/autoconf/headers.m4:663: AC_HEADER_STDC is expanded from...
configure.in:34: the top level
configure.in:53: warning: The macro 'GLIB_GNU_GETTEXT' is obsolete.
configure.in:53: You should run autoupdate.
aclocal.m4:9449: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:9549: AM_GLIB_GNU_GETTEXT is expanded from...
configure.in:53: the top level
configure.in:53: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.in:53: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
aclocal.m4:9134: GLIB_LC_MESSAGES is expanded from...
aclocal.m4:9449: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:9549: AM_GLIB_GNU_GETTEXT is expanded from...
configure.in:53: the top level
configure.in:53: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.in:53: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/headers.m4:89: _AC_CHECK_HEADER_COMPILE is expanded from...
./lib/autoconf/headers.m4:56: AC_CHECK_HEADER is expanded from...
aclocal.m4:9233: GLIB_WITH_NLS is expanded from...
aclocal.m4:9449: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:9549: AM_GLIB_GNU_GETTEXT is expanded from...
configure.in:53: the top level
configure.in:53: warning: The macro 'AC_TRY_LINK' is obsolete.
configure.in:53: You should run autoupdate.
./lib/autoconf/general.m4:2918: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/headers.m4:89: _AC_CHECK_HEADER_COMPILE is expanded from...
./lib/autoconf/headers.m4:56: AC_CHECK_HEADER is expanded from...
aclocal.m4:9233: GLIB_WITH_NLS is expanded from...
aclocal.m4:9449: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:9549: AM_GLIB_GNU_GETTEXT is expanded from...
configure.in:53: the top level
configure.in:53: warning: The macro 'AC_OUTPUT_COMMANDS' is obsolete.
configure.in:53: You should run autoupdate.
./lib/autoconf/status.m4:1025: AC_OUTPUT_COMMANDS is expanded from...
aclocal.m4:9233: GLIB_WITH_NLS is expanded from...
aclocal.m4:9449: GLIB_GNU_GETTEXT is expanded from...
aclocal.m4:9549: AM_GLIB_GNU_GETTEXT is expanded from...
configure.in:53: the top level
configure.in:78: warning: AC_OUTPUT should be used without arguments.
configure.in:78: You should run autoupdate.
Running ./configure --prefix=/nix/store/blblmwx2lhy959yhwmqvlhg5yxdz32z0-cnijfilter2-6.40 --enable-progpath=/nix/store/blblmwx2lhy959yhwmqvlhg5yxdz32z0-cnijfilter2-6.40/bin --datadir=/nix/store/blblmwx2lhy959yhwmqvlhg5yxdz32z0-cnijfilter2-6.40/share --enable-libdir=/var/cache/cups ...
configure: WARNING: unrecognized options: --enable-libdir
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking whether make sets $(MAKE)... (cached) yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for main in -lcups... yes
checking for egrep... (cached) /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for sys/wait.h that is POSIX.1 compatible... yes
checking for arpa/inet.h... yes
checking for fcntl.h... yes
checking for netdb.h... yes
checking for netinet/in.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for sys/socket.h... yes
checking for unistd.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking for size_t... yes
checking for memset... yes
checking for socket... yes
checking for strdup... yes
checking for strrchr... yes
checking for locale.h... yes
checking for LC_MESSAGES... yes
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for libintl.h... yes
checking for ngettext in libc... yes
checking for dgettext in libc... yes
checking for bind_textdomain_codeset... yes
checking for msgfmt... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt
checking for dcgettext... yes
checking if msgfmt accepts -c... yes
checking for gmsgfmt... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt
checking for xgettext... /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/xgettext
checking for catalogs to be installed...  ja de fr zh
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating po/Makefile.in
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing default-1 commands
configure: WARNING: unrecognized options: --enable-libdir
Now type `make' to compile
make  all-recursive
make[1]: Entering directory '/build/source/lgmon3'
Making all in po
make[2]: Entering directory '/build/source/lgmon3/po'
file=./`echo ja | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt -c -o $file ja.po
ja.po:5: warning: header field 'Language' missing in header
file=./`echo de | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt -c -o $file de.po
de.po:5: warning: header field 'Language' missing in header
file=./`echo fr | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt -c -o $file fr.po
fr.po:5: warning: header field 'Language' missing in header
file=./`echo zh | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/9a46kza0vbb4vdsdmpzbdc7f31y7hs0i-gettext-0.22.5/bin/msgfmt -c -o $file zh.po
zh.po:5: warning: header field 'Language' missing in header
make[2]: Leaving directory '/build/source/lgmon3/po'
Making all in src
make[2]: Entering directory '/build/source/lgmon3/src'
gcc -DHAVE_CONFIG_H -I. -I..     -O2 -Wall -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0 -I./common -I/nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2 -MT cnijcomif.o -MD -MP -MF .deps/cnijcomif.Tpo -c -o cnijcomif.o cnijcomif.c
[01m[Kcnijcomif.c:[m[K In function '[01m[KCNIF_GetSerialNum[m[K':
[01m[Kcnijcomif.c:202:9:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 256 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;[m[K]
  202 |         [01;35m[Ks[m[Ktrncpy(dev_uri, (const char *)buffer, CN_DEVICE_URI_LEN);
      |         [01;35m[K^[m[K
mv -f .deps/cnijcomif.Tpo .deps/cnijcomif.Po
gcc -DHAVE_CONFIG_H -I. -I..     -O2 -Wall -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0 -I./common -I/nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2 -MT cnijifusb.o -MD -MP -MF .deps/cnijifusb.Tpo -c -o cnijifusb.o cnijifusb.c
[01m[Kcnijifusb.c:[m[K In function '[01m[KCNIF_USB_IsIvec[m[K':
[01m[Kcnijifusb.c:644:9:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 4096 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;[m[K]
  644 |         [01;35m[Ks[m[Ktrncpy(buffer, info, CN_READ_SIZE);
      |         [01;35m[K^[m[K
[01m[Kcnijifusb.c:[m[K In function '[01m[KCNIF_USB_IsPrinter[m[K':
[01m[Kcnijifusb.c:686:9:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 4096 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;[m[K]
  686 |         [01;35m[Ks[m[Ktrncpy(buffer, info, sizeof(buffer));
      |         [01;35m[K^[m[K
mv -f .deps/cnijifusb.Tpo .deps/cnijifusb.Po
gcc -DHAVE_CONFIG_H -I. -I..     -O2 -Wall -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0 -I./common -I/nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2 -MT cnijlgmon3.o -MD -MP -MF .deps/cnijlgmon3.Tpo -c -o cnijlgmon3.o cnijlgmon3.c
[01m[Kcnijlgmon3.c:[m[K In function '[01m[KCNIF_GetResponse[m[K':
[01m[Kcnijlgmon3.c:987:31:[m[K [01;35m[Kwarning: [m[Kcomparison between pointer and zero character constant [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-pointer-compare-Wpointer-compare]8;;[m[K]
  987 |                 if( supportID [01;35m[K!=[m[K '\0' ) {
      |                               [01;35m[K^~[m[K
[01m[Kcnijlgmon3.c:987:21:[m[K [01;36m[Knote: [m[Kdid you mean to dereference the pointer?
  987 |                 if( [01;36m[Ks[m[KupportID != '\0' ) {
      |                     [01;36m[K^[m[K
[01m[Kcnijlgmon3.c:1059:63:[m[K [01;35m[Kwarning: [m[Kcomparison between pointer and zero character constant [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-pointer-compare-Wpointer-compare]8;;[m[K]
 1059 |                                                 if( supportID [01;35m[K!=[m[K '\0' ){
      |                                                               [01;35m[K^~[m[K
[01m[Kcnijlgmon3.c:1059:53:[m[K [01;36m[Knote: [m[Kdid you mean to dereference the pointer?
 1059 |                                                 if( [01;36m[Ks[m[KupportID != '\0' ){
      |                                                     [01;36m[K^[m[K
[01m[Kcnijlgmon3.c:954:17:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' output truncated before terminating nul copying 28 bytes from a string of the same length [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;[m[K]
  954 |                 [01;35m[Ks[m[Ktrncpy( typePtnMaintenance, SVC_MAINTENANCE, strlen(SVC_MAINTENANCE) );
      |                 [01;35m[K^[m[K
In file included from [01m[K/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/string.h:548[m[K,
                 from [01m[Kcnijlgmon3.h:32[m[K,
                 from [01m[Kcnijlgmon3.c:35[m[K:
In function '[01m[Kstrncat[m[K',
    inlined from '[01m[KCNIF_GetResponse[m[K' at [01m[Kcnijlgmon3.c:929:20[m[K:
[01m[K/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/string_fortified.h:149:10:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin___strncat_chk[m[K' specified bound 16384 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-overflow-Wstringop-overflow=]8;;[m[K]
  149 |   return [01;35m[K__builtin___strncat_chk (__dest, __src, __len,[m[K
      |          [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  150 | [01;35m[K                                  __glibc_objsize (__dest))[m[K;
      |                                   [01;35m[K~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Kcnijlgmon3.c:[m[K In function '[01m[Kmain[m[K':
[01m[Kcnijlgmon3.c:294:9:[m[K [01;35m[Kwarning: [m[K'[01m[K__builtin_strncpy[m[K' specified bound 256 equals destination size [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;[m[K]
  294 |         [01;35m[Ks[m[Ktrncpy(dev_uri, argv[1], CN_DEVICE_URI_LEN);
      |         [01;35m[K^[m[K
mv -f .deps/cnijlgmon3.Tpo .deps/cnijlgmon3.Po
gcc -DHAVE_CONFIG_H -I. -I..     -O2 -Wall -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0 -I./common -I/nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2 -MT cnijifnet.o -MD -MP -MF .deps/cnijifnet.Tpo -c -o cnijifnet.o cnijifnet.c
mv -f .deps/cnijifnet.Tpo .deps/cnijifnet.Po
gcc -DHAVE_CONFIG_H -I. -I..     -O2 -Wall -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0 -I./common -I/nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2 -MT keytext.o -MD -MP -MF .deps/keytext.Tpo -c -o keytext.o keytext.c
[01m[Kkeytext.c:[m[K In function '[01m[KCheckFileExists[m[K':
[01m[Kkeytext.c:57:32:[m[K [01;31m[Kerror: [m[Kimplicit declaration of function '[01m[Kmalloc[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration]8;;[m[K]
   57 |         full_filename = (char*)[01;31m[Kmalloc[m[K( full_filename_len );
      |                                [01;31m[K^~~~~~[m[K
[01m[Kkeytext.c:42:1:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kmalloc[m[K'
   41 | #include "keytext.h"
  +++ |+[32m[K#include <stdlib.h>[m[K
   42 | 
[01m[Kkeytext.c:57:32:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kmalloc[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
   57 |         full_filename = (char*)[01;35m[Kmalloc[m[K( full_filename_len );
      |                                [01;35m[K^~~~~~[m[K
[01m[Kkeytext.c:57:32:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kmalloc[m[K'
[01m[Kkeytext.c:70:9:[m[K [01;31m[Kerror: [m[Kimplicit declaration of function '[01m[Kfree[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration]8;;[m[K]
   70 |         [01;31m[Kfree[m[K(full_filename);
      |         [01;31m[K^~~~[m[K
[01m[Kkeytext.c:70:9:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:70:9:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
[01m[Kkeytext.c:70:9:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:[m[K In function '[01m[KNewStringList[m[K':
[01m[Kkeytext.c:82:52:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kmalloc[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
   82 |         if ( (lpStringListComp = (LPSTRINGLISTCOMP)[01;35m[Kmalloc[m[K(sizeof(STRINGLISTCOMP))) == NULL ) goto Err1;
      |                                                    [01;35m[K^~~~~~[m[K
[01m[Kkeytext.c:82:52:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kmalloc[m[K'
[01m[Kkeytext.c:85:47:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kmalloc[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
   85 |         if ( (lpStringListComp->key = (char *)[01;35m[Kmalloc[m[K(len)) == NULL ) goto Err1;
      |                                               [01;35m[K^~~~~~[m[K
[01m[Kkeytext.c:85:47:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kmalloc[m[K'
[01m[Kkeytext.c:90:48:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kmalloc[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
   90 |         if ( (lpStringListComp->text = (char *)[01;35m[Kmalloc[m[K(len)) == NULL ) goto Err2;
      |                                                [01;35m[K^~~~~~[m[K
[01m[Kkeytext.c:90:48:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kmalloc[m[K'
[01m[Kkeytext.c:101:48:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  101 |         if ( lpStringListComp->text != NULL )  [01;35m[Kfree[m[K( lpStringListComp->text );
      |                                                [01;35m[K^~~~[m[K
[01m[Kkeytext.c:101:48:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:103:47:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  103 |         if ( lpStringListComp->key != NULL )  [01;35m[Kfree[m[K( lpStringListComp->key );
      |                                               [01;35m[K^~~~[m[K
[01m[Kkeytext.c:103:47:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:[m[K In function '[01m[KFreeStringListComp[m[K':
[01m[Kkeytext.c:132:17:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  132 |                 [01;35m[Kfree[m[K( current->key );
      |                 [01;35m[K^~~~[m[K
[01m[Kkeytext.c:132:17:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:136:17:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  136 |                 [01;35m[Kfree[m[K( current->text);
      |                 [01;35m[K^~~~[m[K
[01m[Kkeytext.c:136:17:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:[m[K In function '[01m[KLoadKeyTextList[m[K':
[01m[Kkeytext.c:288:36:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kmalloc[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  288 |         if ( (list = (KeyTextList*)[01;35m[Kmalloc[m[K(sizeof(KeyTextList))) == NULL ) goto Err1;
      |                                    [01;35m[K^~~~~~[m[K
[01m[Kkeytext.c:288:36:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kmalloc[m[K'
[01m[Kkeytext.c:301:9:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  301 |         [01;35m[Kfree[m[K( list );
      |         [01;35m[K^~~~[m[K
[01m[Kkeytext.c:301:9:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
[01m[Kkeytext.c:[m[K In function '[01m[KFreeKeyTextList[m[K':
[01m[Kkeytext.c:311:9:[m[K [01;35m[Kwarning: [m[Kincompatible implicit declaration of built-in function '[01m[Kfree[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch]8;;[m[K]
  311 |         [01;35m[Kfree[m[K( g_key_keytext_list );
      |         [01;35m[K^~~~[m[K
[01m[Kkeytext.c:311:9:[m[K [01;36m[Knote: [m[Kinclude '[01m[K<stdlib.h>[m[K' or provide a declaration of '[01m[Kfree[m[K'
make[2]: *** [Makefile:427: keytext.o] Error 1
make[2]: Leaving directory '/build/source/lgmon3/src'
make[1]: *** [Makefile:416: all-recursive] Error 1
make[1]: Leaving directory '/build/source/lgmon3'
make: *** [Makefile:348: all] Error 2

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

`hplipWithPlugin` also fails to build currently.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`
 
## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

No maintainers listed and the derivation hasn't been touched in a while. No idea who to @.

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/cn/cnijfilter2/package.nix
+++ b/pkgs/by-name/cn/cnijfilter2/package.nix
@@ -36,6 +36,7 @@ stdenv.mkDerivation {
 
   patches = [
     ./patches/get_protocol.patch
+    ./patches/add_missing_import.patch
   ];
 
   # lgmon3's --enable-libdir flag is used soley for specifying in which

--- a/pkgs/by-name/cn/cnijfilter2/patches/add_missing_import.patch
+++ b/pkgs/by-name/cn/cnijfilter2/patches/add_missing_import.patch
@@ -1,0 +1,12 @@
+diff --git a/lgmon3/src/keytext.c b/lgmon3/src/keytext.c
+index 8c15ff8..b80bbe5 100644
+--- a/lgmon3/src/keytext.c
++++ b/lgmon3/src/keytext.c
+@@ -37,6 +37,7 @@
+ #include <unistd.h>
+ #include <libxml/parser.h>	/* Ver.2.80 */
+ #include <string.h>
++#include <stdlib.h>
+ 
+ #include "keytext.h"
+ 

